lib/commands/exam.js

```diff
@@ -268,25 +268,26 @@ module.exports = TestCommand.extend({
     const ui = this.ui;
     const events = config.custom_browser_socket_events || {};
     const testExecutionFileName = `test-execution-${Date.now()}.json`;
+    const browserCount = Object.keys(config.testPage).length;
     const browserExitHandler = function() {
-      const browserCount = Object.keys(config.testPage).length;
+      const browserId = getBrowserId(this.launcher.settings.test_page);
+
       testemEvents.completedBrowsersHandler(
         browserCount,
+        browserId,
         ui,
         commands.get('loadBalance'),
         testExecutionFileName,
         commands.get('writeExecutionFile')
       );
     };
     const browserFailureHandler = function() {
-      if (this.finished) {
-        return;
-      } else if (commands.get('writeExecutionFile')) {
+      if (commands.get('writeExecutionFile')) {
```

Comment:
Removed the check for `this.finish`. This was needed for `should write test-execution json when browser exits` case to fail. 
When a browser exits with timeout, `this.finish` is already set to true. My understanding of the code, was we want to short circuit this function so we don't run `browserExitHandler` because the way it keeps track of browser completion cannot handle duplicates. But with the new implementation, it is not a problem.